core/src/main/java/io/questdb/cairo/CairoEngine.java

```diff
@@ -498,33 +517,43 @@ public TableReader getReaderWithRepair(TableToken tableToken) {
     }
 
     public TableMetadata getSequencerMetadata(TableToken tableToken) {
-        if (!tableToken.isWal()) {
-            return getTableReaderMetadata(tableToken);
-        }
-        verifyTableToken(tableToken);
-        return sequencerMetadataPool.get(tableToken);
-    }
+        return getSequencerMetadata(tableToken, TableUtils.ANY_TABLE_VERSION);
+    }
+
+    /**
+     * Table metadata as seen by the table sequencer. This is the most up-to-date table
+     * metadata, and it can be used to positively confirm column metadata changes immediately after
+     * making them.
+     *
+     * However, this metadata cannot confirm all the changes, one of which is "dedup" flag on a table.
+     * This is a shortcoming and to confirm the "dedup" flag {{@link #getTableMetadata(TableToken, long)}} should
+     * be polled instead. We expect to fix issues like this one in the near future.
+     *
+     * @param tableToken table token
+     * @param desiredVersion version of table metadata used previously if consistent metadata reads are required
+     * @return sequence metadata instance
+     */
+    public TableMetadata getSequencerMetadata(TableToken tableToken, long desiredVersion) {
+
+        assert tableToken.isWal();
 
-    public TableRecordMetadata getSequencerMetadata(TableToken tableToken, long metadataVersion) {
         verifyTableToken(tableToken);
+
         try {
-            final TableRecordMetadata metadata = sequencerMetadataPool.get(tableToken);
-            if (metadataVersion != TableUtils.ANY_TABLE_VERSION && metadata.getMetadataVersion() != metadataVersion) {
-                final TableReferenceOutOfDateException ex = TableReferenceOutOfDateException.of(
-                        tableToken,
-                        metadata.getTableId(),
-                        metadata.getTableId(),
-                        metadataVersion,
-                        metadata.getMetadataVersion()
-                );
-                metadata.close();
-                throw ex;
-            }
-            return metadata;
+            return validateDesiredMetadataVersion(
+                    tableToken,
+                    sequencerMetadataPool.get(tableToken),
+                    desiredVersion
+            );
         } catch (CairoException e) {
             tryRepairTable(tableToken, e);
```

Comment:
No point in trying to repair the table, WAL table sequencer metadata will not be repaired by opening TableWriter.

Reply:
fixed in https://github.com/questdb/questdb/pull/4044/commits/1b26f6db43a998a89649ecb8f103a9fc2740ad43